generator/web/src/main/kotlin/me/kcra/takenaka/generator/web/pages/classPage.kt

```diff
@@ -582,3 +625,43 @@ fun Type.format(remapper: ContextualElementRemapper, isVarargs: Boolean = false)
         else -> className
     }
 }
+
+/**
+ * Formats a modifier integer into a string.
+ *
+ * @param mask the modifier mask (you can get that from the [Modifier] class)
+ * @param classType the type of the class or the type of the class containing the member
+ * @return the modifier string, **may end with a space**
+ */
+fun Int.formatModifiers(mask: Int = 0, classType: ClassType): String = buildString {
+    val mMod = this@formatModifiers and mask
+
+    // remove public and abstract modifiers on interface methods, implicit in some cases
+    // see: https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.4
+    if (classType != ClassType.INTERFACE && (mMod and Opcodes.ACC_PUBLIC) != 0) append("public ")
```

Comment:
this logic would strip `public` modifiers off of interface class modifiers as well